Session.cc

```diff
@@ -214,7 +219,21 @@ void Session::OnRegisterViewer(const CARTA::RegisterViewer& message, uint16_t ic
     ack_message.set_success(success);
     ack_message.set_message(status);
     ack_message.set_session_type(type);
-    ack_message.set_server_feature_flags(CARTA::ServerFeatureFlags::SERVER_FEATURE_NONE);
+
+    uint32_t feature_flags = CARTA::ServerFeatureFlags::REGION_WRITE_ACCESS;
+#ifdef _AUTH_SERVER_
+    feature_flags |= CARTA::ServerFeatureFlags::USER_LAYOUTS;
+    feature_flags |= CARTA::ServerFeatureFlags::USER_PREFERENCES;
+
+    if (!GetLayoutsFromDB(&ack_message)) {
+        // Can log failure here
```

Comment:
should we use the error stream for sending this message to the frontend?

Reply:
That would be good, but I am not sure how to do it.